src/validation/utils.ts

```diff
@@ -1,80 +1,129 @@
 import { ValidationErrorMessage } from '../config'
 import { isIP } from 'net'
 
-export type Validator<V> = (value?: V) => ValidationErrorMessage
+export type Validator<V> = {
+  meta: {
+    min?: number
+    max?: number
+    validation?: string
+  }
+  fn: (value?: V) => ValidationErrorMessage
+}
 export type ValidatorWithParams<P, V> = (param: P, customError?: string) => Validator<V>
 
 // Composes complex validator function that runs each validator in order and returns first occurred error description and skips the rest of validation
 const compose: (f: Validator<string | number>[]) => Validator<number | string> = (
-  validatorFunctions: Validator<number | string>[],
+  validatorFunctions: Validator<number | string>[]
 ) => {
-  return (value) => {
-    for (const validator of validatorFunctions) {
-      const errorText = validator(value)
-      if (errorText?.length) {
-        return errorText
+
+  let min;
+  let max;
+  let validation = ''
+  
+  for (let i = 0; i < validatorFunctions.length; i++) {
+    const curValidator = validatorFunctions[i]
+    if (curValidator.meta.min !== undefined) { min = curValidator.meta.min }
+    if (curValidator.meta.max !== undefined) { max = curValidator.meta.max }
+    
+    validation += curValidator.meta.validation
+    if (i < validatorFunctions.length - 1) { validation += ', ' }
+  }
```

Comment:
```suggestion
  const meta: ValidatorMeta = {}
  let details: string[] = []
  for (const validator of validators) {
    if (validator.meta.min) meta.min = validator.meta.min
    if (validator.meta.max) meta.max = validator.meta.max
    if (validator.meta.details) details.push(validator.meta.details)
  }
  meta.details = details.join(', ')
```

Reply:
thanks I like this much better! 

Only thing I'm keeping is "curValidator.meta.min !== undefined"

If the min is 0 it won't be truthy, hence checking if it is undefined. I could change the if statement checking for the max since it is never 0, but I kind of like the verbosity in this case. what do you think?